httpReq.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * The response body is automatically parsed according
+ * to the response content type.
+ *
+ * @param {string} method             - the http method, e.g. GET, PUT
+ * @param {string} path               - the url path that gets appended to baseUrl
+ * @param {*} options.payload         - optional payload to be send with req
+ * @param {string} options.baseUrl    - base for url, defaults to dw api domain
+ * @param {string} options.contentType - base for url, defaults to dw api domain
+ *
+ * @returns {Promise} fetch promise to the parsed response body
+ *
+ * @example
+ *  import httpRequest from '@datawrapper/shared/httpRequest';
+ *  let res = await httpRequest('POST', '/v3/charts', {
+        payload: {
+            title: 'My new chart'
+        }
+    })
+    res = await httpRequest('/v3/charts', {
+        method: 'POST',
+        payload: {
+            title: 'My new chart'
+        }
+    })
+ */
+/* globals dw */
+export default httpReq;
+
+httpReq.get = httpReqVerb('get');
+httpReq.patch = httpReqVerb('patch');
+httpReq.delete = httpReqVerb('delete');
+httpReq.put = httpReqVerb('put');
+httpReq.post = httpReqVerb('post');
+
+function httpReqVerb(method) {
+    return (path, options) => {
+        if (options && options.method.toLowerCase() != method) {
+            throw new Error(`Invalid method "${options.method}" provided in ${method}() call`);
+         }
+         return httpReq(path, {...options, method});
+    };
+}
+
+function httpReq(path, options = {}) {
+    const { payload, baseUrl, raw, ...opts } = {
+        payload: null,
+        raw: false,
+        method: 'get',
+        baseUrl: `//${dw.backend.__api_domain}`,
+        mode: 'cors',
+        credentials: 'include',
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        ...options
+    };
```

Comment:
This is a fun one with object rest spread (and `Object.assign`). If I now call `httpReq` like this:
```js
httpReq('/', {
  headers: { 'Accept-Encoding': 'gzip' }
})
```
`options.headers` will overwrite the default `Content-Type` header. If you want this to be possible you need to change order around:

```diff
const { payload, baseUrl, raw, ...opts } = {
    payload: null,
    raw: false,
    method: 'get',
    baseUrl: `//${dw.backend.__api_domain}`,
    mode: 'cors',
    credentials: 'include',
+   ...options,
    headers: {
        'Content-Type': 'application/json',
+       ...options.headers
    }
-   ...options
};

Reply:
👍